Nataly_Bondarenko/HW10/task2_class_worker.py

```diff
@@ -0,0 +1,179 @@
+class Worker:
+    def __init__(self, first_name: str, last_name: str, age: int, position: str, salary: float):
+        self.__first_name = first_name
+        self.__last_name = last_name
+        self.__age = age
+        self.__position = position
+        self.__salary = salary
+
+    @property
+    def first_name(self):
+        """
+        Method  getting the worker's name
+        """
+        return self.__first_name
+
+    @first_name.setter
+    def first_name(self, new_first_name: str):
+        """
+        method allows to change the name of the worker
+        """
+        if len(new_first_name):
+            self.__first_name = new_first_name.capitalize()
+        else:
+            print('Empty value is not supported for worker name')
+
+    @first_name.deleter
+    def first_name(self):
+        """
+        method will protect code against class design break
+        """
+        pass
+
+    @property
+    def last_name(self):
+        """
+        Method  getting the worker's last name
+        """
+        return self.__last_name
+
+    @last_name.setter
+    def last_name(self, new_last_name: str):
+        if self.__last_name == new_last_name:
+            print(f'{new_last_name} - this last name already exists for this worker')
+        elif not len(new_last_name):
+            print('Empty value is not supported for worker last name')
+        else:
+            self.__last_name = new_last_name.capitalize()
+
+    @last_name.deleter
+    def last_name(self):
+        """
+        method will protect code against class design break
+        """
+        pass
+
+    @property
+    def age(self):
+        """
+        Method  getting the worker's age
+        """
+        return self.__age
+
+    @age.setter
+    def age(self, age_verification):
+        """
+        Method verified that worker's age is more than 18 years old
+        """
+        if age_verification < 18:
+            print(f'Sorry, people under 18 years old can not works in our company')
+        else:
+            print('We are glad that you are part of our company')
```

Comment:
it is not a setter. You have never setted any value

Reply:
changed this verification to general method